controllers/platformnetwork_controller.go

```diff
@@ -1248,6 +1248,13 @@ func (r *PlatformNetworkReconciler) Reconcile(ctx context.Context, request ctrl.
 		return reconcile.Result{}, err
 	}
 
+	// Restore the PlatformNetwork status
+	if r.checkRestoreInProgress(instance) {
+		if err := r.RestorePlatformNetworkStatus(instance); err != nil {
+			return reconcile.Result{}, err
+		}
```

Comment:
What if err==nil?
We would like to return err, whatever be the case, it could either be nil or be NewResourceStatusDependency error.

Reply:
Changes are done